src/remote-config/remote-config-api-client.ts

```diff
@@ -23,17 +23,26 @@ import * as validator from '../utils/validator';
 
 // Remote Config backend constants
 const FIREBASE_REMOTE_CONFIG_V1_API = 'https://firebaseremoteconfig.googleapis.com/v1';
-const FIREBASE_REMOTE_CONFIG_HEADERS = {
+const FIREBASE_REMOTE_CONFIG_GET_HEADERS = {
   'X-Firebase-Client': 'fire-admin-node/<XXX_SDK_VERSION_XXX>',
   // There is a known issue in which the ETag is not properly returned in cases where the request
   // does not specify a compression type. Currently, it is required to include the header
   // `Accept-Encoding: gzip` or equivalent in all requests.
   // https://firebase.google.com/docs/remote-config/use-config-rest#etag_usage_and_forced_updates
   'Accept-Encoding': 'gzip',
 };
+const FIREBASE_REMOTE_CONFIG_PUT_HEADERS = {
+  'X-Firebase-Client': 'fire-admin-node/<XXX_SDK_VERSION_XXX>',
+  // There is a known issue in which the ETag is not properly returned in cases where the request
+  // does not specify a compression type. Currently, it is required to include the header
+  // `Accept-Encoding: gzip` or equivalent in all requests.
+  // https://firebase.google.com/docs/remote-config/use-config-rest#etag_usage_and_forced_updates
+  'Accept-Encoding': 'gzip',
+  'If-Match': '',
```

Comment:
Looks unnecessary.

Reply:
This was added as we need to set the `etag` at runtime. I was getting the following error without adding the property here.
```
Element implicitly has an 'any' type because expression of type '"If-Match"' can't be used to index type '{ 'X-Firebase-Client': string; 'Accept-Encoding': string; }'.
```

Reply:
Define the type of this object to be `{[key: string]: any}` or similar to avoid the TS error.